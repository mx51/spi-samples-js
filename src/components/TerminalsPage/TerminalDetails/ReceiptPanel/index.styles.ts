import { createStyles, makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3),
    },
  })
);
export default useStyles;
