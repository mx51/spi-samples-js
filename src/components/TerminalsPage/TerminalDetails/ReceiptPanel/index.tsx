import React from 'react';
import { Box } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { useSelector } from 'react-redux';
import selectedTerminalIdSelector from '../../../../redux/reducers/SelectedTerminalSlice/selectedTerminalSliceSelector';
import { ITerminalProps } from '../../../../redux/reducers/TerminalSlice/interfaces';
import { terminalInstance } from '../../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import useStyles from './index.styles';

export default function ReceiptPanel(): React.ReactElement {
  const selectedTerminal = useSelector(selectedTerminalIdSelector);
  const currentTerminal = useSelector(terminalInstance(selectedTerminal)) as ITerminalProps;

  const classes = useStyles();
  return (
    <Box className={classes.receiptBoxWrapper}>
      <Box className={classes.receiptBox}>
        <Typography variant="h6" component="h1">
          Receipt
        </Typography>
        <pre>{currentTerminal?.txFlow?.response?.data.merchantReceipt}</pre>
      </Box>
    </Box>
  );
}
