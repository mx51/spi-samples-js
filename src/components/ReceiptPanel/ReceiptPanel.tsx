import React, { useState } from 'react';
import { Box, Button } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { IReceiptPanel } from '../TerminalsPage/TerminalDetails/interfaces';
import { useReceiptPanelStyles } from './index.styles';
import { ReactComponent as TickIcon } from '../../images/TickIcon.svg';
import { ReactComponent as CopyIcon } from '../../images/CopyIcon.svg';

export default function ReceiptPanel({ children, title, css, textReceipt }: IReceiptPanel): React.ReactElement {
  const classes = useReceiptPanelStyles();
  const [copySuccess, setCopySuccess] = useState<string>('');

  const copyToClipboard = async (receipt: string) => {
    try {
      await navigator.clipboard.writeText(receipt);
      setCopySuccess('Success');
      setTimeout(() => {
        setCopySuccess('');
      }, 1000);
    } catch (err) {
      console.log('Failed to copy: ', err);
    }
  };

  return (
    <Grid className={css}>
      <Box className={classes.box}>
        <Grid className={classes.gridContainer}>
          <Grid item md={12} className={classes.receiptGrid}>
            <Typography variant="h6" component="h1">
              {title}
            </Typography>
            {children}
          </Grid>
          <Grid item md={12} className={classes.buttonGrid}>
            {textReceipt && (
              <Button className={classes.button} onClick={() => copyToClipboard(textReceipt)}>
                {copySuccess ? <TickIcon /> : <CopyIcon />}
                Copy Receipt
              </Button>
            )}
          </Grid>
        </Grid>
      </Box>
    </Grid>
  );
}
