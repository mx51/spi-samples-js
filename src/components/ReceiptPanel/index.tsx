import React from 'react';
import { Box } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { IReceiptPanel } from '../TerminalsPage/TerminalDetails/interfaces';

export default function ReceiptPanel({ children, title, css }: IReceiptPanel): React.ReactElement {
  return (
    <Grid className={css}>
      <Box>
        <Typography variant="h6" component="h1">
          {title}
        </Typography>
        {children}
      </Box>
    </Grid>
  );
}
