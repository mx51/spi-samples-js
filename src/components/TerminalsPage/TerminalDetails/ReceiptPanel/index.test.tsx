import React from 'react';
import { cleanup } from '@testing-library/react';
import ReceiptPanel from '.';
import mockWithRedux from '../../../../utils/tests/common';

describe('Test <ReceiptPanel />', () => {
  afterEach(cleanup);

  test('should match ReceiptPanel snapshot test', () => {
    // Arrange
    const container = mockWithRedux(<ReceiptPanel />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
