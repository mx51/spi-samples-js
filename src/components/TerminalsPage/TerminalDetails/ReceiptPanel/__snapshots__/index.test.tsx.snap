// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ReceiptPanel /> should match ReceiptPanel snapshot test 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-receiptBoxWrapper-1"
  >
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-receiptBox-2"
    >
      <h1
        class="MuiTypography-root MuiTypography-h6"
      >
        Receipt
      </h1>
      <pre />
    </div>
  </div>
</div>
`;
